main.tf

```diff
@@ -0,0 +1,167 @@
+  // separate resource below.
+  remove_default_node_pool = "true"
+  initial_node_count       = 1
+
+  // Configure various addons
+  addons_config {
+    // Disable the Kubernetes dashboard, which is often an attack vector. The
+    // cluster can still be managed via the GKE UI.
+    kubernetes_dashboard {
+      disabled = true
+    }
+
+    // Enable network policy (Calico)
+    network_policy_config {
+      disabled = false
+    }
+  }
+
+  // Enable workload identity
+  workload_identity_config {
+    identity_namespace = format("%s.svc.id.goog", var.project)
+  }
+
+  // Disable basic authentication and cert-based authentication.
+  // Empty fields for username and password are how to "disable" the
+  // credentials from being generated.
+  master_auth {
+    username = ""
+    password = ""
+
+    client_certificate_config {
+      issue_client_certificate = "false"
+    }
+  }
+
+  // Enable network policy configurations (like Calico) - for some reason this
+  // has to be in here twice.
+  network_policy {
+    enabled = "true"
+  }
+
+  // Allocate IPs in our subnetwork
+  ip_allocation_policy {
+    use_ip_aliases                = true
+    cluster_secondary_range_name  = google_compute_subnetwork.subnetwork.secondary_ip_range.0.range_name
+    services_secondary_range_name = google_compute_subnetwork.subnetwork.secondary_ip_range.1.range_name
+  }
+
+  // Specify the list of CIDRs which can access the master's API
+  master_authorized_networks_config {
+    cidr_blocks {
+      display_name = "bastion"
+      cidr_block   = format("%s/32", google_compute_instance.bastion.network_interface.0.network_ip)
+    }
+  }
+  // Configure the cluster to have private nodes and private control plane access only
+  private_cluster_config {
+    enable_private_endpoint = "true"
+    enable_private_nodes    = "true"
+    master_ipv4_cidr_block  = "172.16.0.16/28"
+  }
+
+  // Allow plenty of time for each operation to finish (default was 10m)
+  timeouts {
+    create = "30m"
+    update = "30m"
+    delete = "30m"
+  }
+
+  depends_on = [
+    google_project_service.service,
+    google_project_iam_member.service-account,
+    google_project_iam_member.service-account-custom,
+    google_compute_router_nat.nat,
+  ]
+
+}
+
+// A dedicated/separate node pool where workloads will run.  A regional node pool
+// will have "node_count" nodes per zone, and will use 3 zones.  This node pool
+// will be 3 nodes in size and use a non-default service-account with minimal
+// Oauth scope permissions.
+resource "google_container_node_pool" "private-np-1" {
+  provider = google-beta
+
+  name       = "private-np-1"
+  location   = var.zone
+  cluster    = google_container_cluster.cluster.name
+  node_count = "3"
+
+  // Repair any issues but don't auto upgrade node versions
+  management {
+    auto_repair  = "true"
+    auto_upgrade = "false"
+  }
+
+  node_config {
+    machine_type = "n1-standard-2"
```

Comment:
parameterize machine_type using variables.tf

Reply:
do you want the machine_type to be started with n1-standard-1 or n1-standard-2?

Reply:
2 x n1-highmem-2